Sources/MongoSwift/MongoClient.swift

```diff
@@ -14,24 +14,32 @@ public struct ClientOptions: CodingStrategyProvider {
     /// be used.
     public let readConcern: ReadConcern?
 
-    /// Specifies a ReadPreference to use for the client.
-    public let readPreference: ReadPreference?
-
     /// Specifies a WriteConcern to use for the client. If one is not specified, the server's default write concern
     /// will be used.
     public let writeConcern: WriteConcern?
 
+    // swiftlint:disable redundant_optional_initialization
```

Comment:
It would be pretty easy to implement, but I didn't see any tests that specified one so I didn't bother. I was mainly thinking of only us using this `Decodable` behavior for testing, but this and your other point bring up the question of whether _users_ would ever want to read the various options types from disk or not. I originally didn't think so, but there could be potentially some value in that. It could possibly allow for on the fly configuration without stopping their backend applications and recompiling.

For the purposes of this PR, I think we can leave it as-is. I filed [a ticket](https://jira.mongodb.org/browse/SWIFT-492) for tracking the future work of making all options `Codable`.